plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java

```diff
@@ -81,6 +84,59 @@ protected DeployImagePage(IWizard wizard, IDeployImagePageModel model) {
 		this.model = model;
 	}
 
+	/**
+	 * Callback that gets called when this page is going to be deactivated. This
+	 * is the chance to do a one-time remote call to the selected Docker daemon
+	 * to check if an image with the given name exists in the local cache or in
+	 * the remote registry (Docker Hub). This kind of validation is a
+	 * long-running process, so it should not be performed during the field
+	 * validation.
+	 * 
+	 * @param progress
+	 *            the direction that the wizard is moving: backwards/forwards
+	 * @param event
+	 *            the page changing event that may be use to veto the change
+	 * @param dbc
+	 *            the current data binding context
+	 */
+	protected void onPageWillGetDeactivated(Direction progress, PageChangingEvent event, DataBindingContext dbc) {
+		/**
+		 * Inner class to perform the image search in the selected Docker daemon cache or on the remote registry. 
+		 */
+		class ImageValidator implements IRunnableWithProgress {
```

Comment:
any special reason to use the wizard runnable vs WizardUtils#runInWizard(job) - that's what we use consistently across openshift for the ability to reuse jobs if possible?